### PR TITLE
Exams Veronica Pylypchuk IM-33

### PR DESCRIPTION
**Pull request by namanica**

### CODE REVIEW

**Comment by DavidkJun** [comment]
I agree with all the changes after the review.

**Comment by yur-ochka** [comment]
I have reviewed the changes and can totally agree with them!

**Comment by Adaptolog** [comment]
Great job on the code! Overall, it's clean and well-organized.

**Comment by Kiryaoo** [comment]
Did a good job with the code.

**Comment by alomj** [comment]
you have done a great job!

**Comment by thonly-zukich** [comment]
Fantastic job on GitHub! Your code is impressively neat and very user-friendly. Excellent work!

**Comment by nikama-s** [comment]
Fantastic job, i agree with everething

**Comment by nevsetakpohano** [comment]
It brought me only joy, reviewing your changes. Such a great job!

**Comment by D-Vasylchenko** [comment]
Looked through everything and really liked it!

File: Tasks/day.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-let D = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
 
-_parse_day_ = (s) => {
-  let i;
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
+const parseDay = (day) => {
+  for (let i = 0; i < days.length; i++) {
+    if (day.startsWith(days[i].toLowerCase())) {
```
Comment (tshemsedinov) [inline Tasks/day.js:8]: Extract invariants

File: Tasks/duplicate.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 
-duplicate = (value, N) => {
-  if (N <= 0) return [];
-  else {
-    res = [];
-    for (let i = 0; i < N; i++) {
+const duplicateValue = (value, number) => {
+  if (number <= 0) {
+    return [];
+  } else {
```
Comment (tshemsedinov) [inline Tasks/duplicate.js:7]: Do not need else

File: Tasks/month.js [hunk 3]
```diff
@@ -15,12 +16,11 @@ Months = [
   'dec',
 ];
 
-Month = (s) => {
-  l = Months.length;
-  for (let i = 0; i < l; i++) {
-    if (s.toLowerCase().startsWith(Months[i])) return i + 1;
+const getMonthIndex = (month) => {
+  for (let i = 0; i < months.length; i++) {
+    if (month.toLowerCase().startsWith(months[i])) return i + 1;
```
Comment (tshemsedinov) [inline Tasks/month.js:21]: Long expression, decomose

File: Tasks/month.js [hunk 4]
```diff
@@ -15,12 +16,11 @@ Months = [
   'dec',
 ];
 
-Month = (s) => {
-  l = Months.length;
-  for (let i = 0; i < l; i++) {
-    if (s.toLowerCase().startsWith(Months[i])) return i + 1;
+const getMonthIndex = (month) => {
+  for (let i = 0; i < months.length; i++) {
```
Comment (tshemsedinov) [inline Tasks/month.js:20]: For..of

File: Tasks/reverse.js [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
-    DATA[v1] = _;
-    delete DATA[_];
-  }, 1000);
-  return DATA;
+const reverseObj = (obj) => {
+  //second variant using object methods
+  //return Object.fromEntries(Object.entries(obj).map(([key,value])=>[value,key]));
+  Object.keys(obj).forEach((key) => {
+    obj[obj[key]] = key;
+    delete obj[key];
```
Comment (tshemsedinov) [inline Tasks/reverse.js:9]: Implement without delete

**Pull request closed without merging**